hammr/commands/image/image.py

```diff
@@ -108,6 +108,148 @@ def help_list(self):
         doParser = self.arg_list()
         doParser.print_help()
 
+    def arg_info(self):
+        do_parser = ArgumentParser(prog=self.cmd_name + " list", add_help=True,
+                                  description="Retrieve detailed information of a generated image")
+
+        mandatory = do_parser.add_argument_group("mandatory arguments")
+
+        mandatory.add_argument('--id', dest='id', type=str, required=True,
+                               help="The unique identifier of the image to retrieve")
+
+        return do_parser
+
+    def do_info(self, args):
+        try:
+            do_parser = self.arg_info()
+            do_args = do_parser.parse_args(shlex.split(args))
+            if not do_args:
+                raise ArgumentParserError("No arguments")
+
+            images = self.get_all_images()
+
+            info_image = self.get_image(images, do_args.id)
+            if info_image is None:
+                printer.out("The image with id \"" + do_args.id + "\" doesn't exist.")
+                return 0
+
+            printer.out("Informations about [" + info_image.name + "] :")
+
+            self.do_info_draw_general(info_image)
+            self.do_info_draw_generation(info_image)
+            self.do_info_draw_publication(info_image)
+            return 0
+
+        except ArgumentParserError as e:
+            printer.out("In Arguments: " + str(e), printer.ERROR)
+            self.help_info()
+        except Exception as e:
+            return handle_uforge_exception(e)
+
+    def do_info_draw_general(self, info_image):
+        table = Texttable(0)
+        table.set_cols_dtype(["a", "t"])
+        table.set_cols_align(["l", "l"])
+
+        table.add_row(["Name", info_image.name])
+        table.add_row(["Format", info_image.targetFormat.name])
+        table.add_row(["Id", info_image.dbId])
+        table.add_row(["Version", info_image.version])
+        table.add_row(["Revision", info_image.revision])
+        table.add_row(["Uri", info_image.uri])
+
+        self.do_info_draw_source(info_image.parentUri, table)
+
+        table.add_row(["Created", info_image.created.strftime("%Y-%m-%d %H:%M:%S")])
+        table.add_row(["Size", size(info_image.fileSize)])
+        table.add_row(["Compressed", "Yes" if info_image.compress else "No"])
+
+        format_name = info_image.targetFormat.format.name
+        if format_name == "docker" or format_name == "openshift":
+            table.add_row(["RegisteringName", info_image.registeringName])
+            table.add_row(["Entrypoint", info_image.entrypoint])
+
+        print table.draw() + "\n"
+
+    def do_info_draw_source(self, parent_uri, table):
+        appliances = self.api.Users(self.login).Appliances.Getall()
```

Comment:
Isn't it possible to use the Appliances.Get method instead? (appliance id could be extracted from appliance URI)? It would avoid looping on appliances afterwards.

Reply:
Good idea, thank you